### PR TITLE
Backward tests

### PR DESCRIPTION
**Pull request by Bartzi**
add some more tests for the graph backward function

### CODE REVIEW

File: length/functions/add.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from length.function import Function
+
+
+class Add(Function):
```
Comment (Jopyth) [inline length/functions/add.py:4]: Should we also add a test for `add`?

File: length/tests/graph_tests/test_graph_backward.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+def test_graph_backward_with_layers():
+    # use a fully connected layer and have a look whether the backward pass distributes the gradients correctly
+    data = np.random.uniform(-1, 1, (2, 2)).astype(constants.DTYPE)
+    labels = np.array([1, 1], dtype=np.int32)
+
+    fc_layer = FullyConnected(2, 2)
+    fc_layer.weights[...] = np.zeros_like(fc_layer.weights)
+    fc_layer.bias[...] = np.array([-10, 10])
+
+    def run_forward(x, t):
```
Comment (Jopyth) [inline length/tests/graph_tests/test_graph_backward.py]: What is `x` and `t`? Maybe explicitly call them `input` and `labels`?

File: length/functions/add.py (hunk 1 above)
Comment (Bartzi) [reply length/functions/add.py:4]: Yes, I did -> please review again

**Review by Jopyth — APPROVED** [review]
:eagle:

**Pull request merged**